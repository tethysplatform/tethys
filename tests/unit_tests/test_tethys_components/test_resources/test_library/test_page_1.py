def test(lib):
    lib.register("@monaco-editor/react", "me", default_export="Editor")

    return lib.html.div()(
        lib.bs.Container(fluid=True, style=lib.utils.Props(height="calc(100vh - 75px"))(
            lib.bs.Row()(lib.bs.Button()("Render")),
            lib.bs.Row()(
                lib.bs.Col(style=lib.Style(width="50vw"))(
                    lib.me.Editor(
                        height="80vh",
                        language="python",
                        theme="vs-dark",
                        value="",
                        options=lib.Props(
                            inlineSuggest=True, fontSize="16px", formatOnType=True
                        ),
                    )
                ),
                lib.bs.Col(style=lib.Style(width="50vw"))(
                    lib.html.iframe(
                        style=lib.Style(height="100%", width="100%"),
                        src="/apps/component-app-playground/render",
                        title="Result",
                    )
                ),
            ),
            lib.bs.Row()(lib.bs.Button()("Render")),
        )
    )
