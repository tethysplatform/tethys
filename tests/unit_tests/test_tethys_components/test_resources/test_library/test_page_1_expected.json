{"tagName": "div", "children": [{"tagName": "Container", "attributes": {"fluid": true, "style": {"height": "calc(100vh - 75px"}}, "importSource": {"source": "test_page_1.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "Row", "importSource": {"source": "test_page_1.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "Button", "importSource": {"source": "test_page_1.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": ["Render"]}]}, {"tagName": "Row", "importSource": {"source": "test_page_1.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "Col", "attributes": {"style": {"width": "50vw"}}, "importSource": {"source": "test_page_1.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "Editor", "attributes": {"height": "80vh", "language": "python", "theme": "vs-dark", "value": "", "options": {"inlineSuggest": true, "fontSize": "16px", "formatOnType": true}}, "importSource": {"source": "test_page_1.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}, {"tagName": "Col", "attributes": {"style": {"width": "50vw"}}, "importSource": {"source": "test_page_1.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "iframe", "attributes": {"style": {"height": "100%", "width": "100%"}, "src": "/apps/component-app-playground/render", "title": "Result"}}]}]}, {"tagName": "Row", "importSource": {"source": "test_page_1.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": [{"tagName": "Button", "importSource": {"source": "test_page_1.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}, "children": ["Render"]}]}]}]}